File: su/src/main/java/org/sqlbuilder/su/SuWordResolver.java
package org.sqlbuilder.su;

import org.sqlbuilder.common.DeSerialize;
import org.sqlbuilder.common.Resolver;

import java.io.File;
import java.io.IOException;

public class SuWordResolver extends Resolver<String, Integer>
{
	public SuWordResolver(final String ser) throws IOException, ClassNotFoundException
	{
		super(DeSerialize.deserialize(new File(ser)));
	}

	public static void main(String[] args) throws IOException, ClassNotFoundException
	{
		var r = new SuWordResolver(args[0]);
		var id = r.apply("airport");
		System.out.println(id);
	}
}

File: su/src/main/java/org/sqlbuilder/su/SuWordResolver.kt
package org.sqlbuilder.su

import org.sqlbuilder.common.DeSerialize.deserialize
import org.sqlbuilder.common.Resolver
import java.io.File

typealias SuWordResolvable = String
typealias SuWordResolved = Int

class SuWordResolver(ser: String) : Resolver<SuWordResolvable, SuWordResolved>(deserialize<Map<SuWordResolvable, SuWordResolved>>(File(ser)))
